GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -462,12 +462,22 @@ + (GIDSignIn *)sharedInstance {
 - (id)initPrivate {
   id<GIDKeychainHandler> keychainHandler = [[GIDKeychainHandler alloc] init];
   id<GIDHTTPFetcher> httpFetcher = [[GIDHTTPFetcher alloc] init];
+  
+  NSString *authorizationEnpointURL = [NSString stringWithFormat:kAuthorizationURLTemplate,
+      [GIDSignInPreferences googleAuthorizationServer]];
+  NSString *tokenEndpointURL = [NSString stringWithFormat:kTokenURLTemplate,
+      [GIDSignInPreferences googleTokenServer]];
+  OIDServiceConfiguration *appAuthConfiguration = [[OIDServiceConfiguration alloc]
+      initWithAuthorizationEndpoint:[NSURL URLWithString:authorizationEnpointURL]
+                      tokenEndpoint:[NSURL URLWithString:tokenEndpointURL]];
```

Comment:
The initialization seems pretty static, so it can be placed anywhere essentially. And it feels like it should be just an implementation detail of the proposed `GIDAuthorizationFlowProcessor` unless it will still be needed outside `GIDAuthorizationFlowProcessor`.

Reply:
It will also be used in GIDSignIn.m here.

[GIDAuthStateMigration migrateIfNeededWithTokenURL:_appAuthConfiguration.tokenEndpoint
                                          callbackPath:kBrowserCallbackPath
                                          keychainName:kGTMAppAuthKeychainName
                                        isFreshInstall:isFreshInstall];

We can also consider generating the token endpoint again to get rid of `_appAuthConfiguration` but I feel that makes our maintenance harder.

Reply:
It looks like `GIDAuthStateMigration` only needs the token URL. In that case, I'd suggest putting the token URL in a common utility class like `GIDSignInPreferences`, and move initialization of `OIDServiceConfiguration` into `GIDAuthorizationFlowProcessor`.

Reply:
Done.